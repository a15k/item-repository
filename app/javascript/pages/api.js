import { React } from '../helpers/react';
import styled from 'styled-components';
import Interactions from './interactions';
import Tokens from './tokens';

const Divider = styled.hr`
border-width: 5px;
border-radius: 3px;
`;


const Api = () => (
  <div>
    <Tokens />
    <Divider />
    <Interactions />
  </div>
);

export default Api;
