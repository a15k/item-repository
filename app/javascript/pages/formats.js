import { withRouter } from 'react-router-dom';
import { ListGroup, ListGroupItem } from 'reactstrap';
import { React, PropTypes, observer, action } from '../helpers/react';
import styled from 'styled-components';
import Format from '../models/format';
import Button from '../components/button';
import ActivityIndicator from '../components/activity-indicator';
import ModelCollection from '../models/model-collection';
import NotFound from './not-found';

const FormatItem = styled(ListGroupItem)`
  display: flex;
  justify-content: space-between;
`;

@withRouter
@observer
export default class Formats extends React.Component {

  static propTypes = {
    collection: PropTypes.instanceOf(ModelCollection),
    match: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  };

  static defaultProps = {
    collection: Format.collection,
  };

  componentDidMount() {
    if (!this.props.collection.api.isFetchedOrFetching) {
      this.props.collection.fetch();
    }
  }

  @action.bound onDisplay(ev) {
    this.props.history.push(`/formats/${ev.currentTarget.dataset.id}`);
  }

  @action.bound onDisplayAll() {
    this.props.history.push('/formats');
  }

  render() {
    const { collection, match: { params: { id } } } = this.props;
    if (collection.api.isPending) {
      return <ActivityIndicator />;
    }

    if (id) {
      const format = collection.get(id);
      if (!format) {
        return <NotFound heading="Format not found" />;
      }
      return (
        <div>
          <h3>
            <Button icon="chevronLeft" onClick={this.onDisplayAll}>Display All</Button>
            {format.name}
          </h3>
          <p>{format.specification}</p>
        </div>
      );
    }
    return (
      <div className="formats-page">
        <ListGroup>
          {collection.array.map(f =>
            <FormatItem key={f.id}>
              <span>{f.name}</span>
              <span>{f.id}</span>
              <Button icon="chevronRight" data-id={f.id} onClick={this.onDisplay} />
            </FormatItem>)}
        </ListGroup>
      </div>
    );
  }
}
