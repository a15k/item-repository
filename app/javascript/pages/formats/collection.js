import { action, observable } from 'mobx';
import Format from '../../models/format';
import lazyGetter from '../../helpers/lazy-getter.js';
import ModelApi from '../../models/api';

export default class SearchCollection {

  @observable formats = [];
  @lazyGetter api = new ModelApi(this, 'formats');

  @observable isLoading = false;

  @action search(q) {
    this.isLoading = true;
    return this.api.request({ model: this, query: { q } })
      .then(() => {
        this.isLoading = false;
      });
  }

  fromJSON(models) {
    this.formats.replace(
      models.map(a => new Format(a))
    );
  }

  @action.bound fetchById(id) {
    this.isLoading = true;
    return Format
      .fetch({ id })
      .then((format) => {
        this.formats.replace([format]);
        this.isLoading = false;
      });
  }

  map(fn) {
    return this.formats.map(fn);
  }
}
