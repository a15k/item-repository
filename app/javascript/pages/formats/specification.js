import iframeResizer from 'iframe-resizer/js/iframeResizer';
import { React, PropTypes, observer } from '../../helpers/react';
import Format from '../../models/format';

@observer
export default class FormatSpecification extends React.Component {
  static propTypes = {
    format: PropTypes.instanceOf(Format).isRequired,
  }
  componentDidMount() {
    this.iframe = iframeResizer({}, `#format-specification-${this.props.format.id}`)[0];
  }
  componentWillUnmount() {
    if (this.iframe) {
      this.iframe.iFrameResizer.close();
    }
  }
  render() {
    const { format } = this.props;
    return (
      <iframe
        srcDoc={format.specification_document}
        id={`format-specification-${format.id}`}
        src={`/assessment/format/${format.id}`}
        style={{ width: '1px', minWidth: '100%', border: '0px' }}
      />
    );
  }
}
