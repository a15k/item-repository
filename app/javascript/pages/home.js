import { React, observer } from '../helpers/react';
import User from '../models/user';
import { map } from 'lodash';
import { toSentence } from '../helpers/string';
import Config from '../models/config';

const NonMemberInfo = () => (
  <React.Fragment>
    <p>You are currently not a part of a network member.  If you think you should be,
      please contact the appropriate staff at your organization and they can add you
      to your organization’s member.
    </p>
    <p>
      If your organization is not currently a network member, and you are interested in joining the network, please contact us at <a href="mailto:info@a15k.org">info@a15k.org</a>.
    </p>
  </React.Fragment>
);


const MemberInfo = () => (
  <React.Fragment>
    <p>
      You are a user in the {Config.member_name} member.  Your member’s
      power {Config.member_power_users.length > 1 ? 'users are ' : 'user is '}
      {toSentence(map(Config.member_power_users, 'name'))}. Please contact them
      with questions about your organization’s membership in the network.
    </p>
    <p>
      You can access network functionality through the menu dropdowns in the upper
      right.  Have questions about how to use the network?  Take a look at the
      material on our <a href="http://develop.a15k.org">developer portal</a> or
      contact us at <a href="mailto:info@a15k.org">info@a15k.org</a>.
    </p>
  </React.Fragment>
);

@observer
export default class HomePage extends React.Component {

  render() {
    return (
      <div className="homepage">
        <h1>The Assessment Network</h1>
        <p>
          The Assessment Network (a15k), established by Rice University’s OpenStax initiative, is a community of member organizations working through sustainable business models to pool assessment items and the knowledge accrued about them.
        </p>

        {User.isMember && <MemberInfo />}
        {!User.isMember && <NonMemberInfo />}
      </div>
    );
  }

}
