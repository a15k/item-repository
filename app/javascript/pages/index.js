import Home from './home';
import Api from './api';
import Search from './search';
import NotFound from './not-found';
import Users from './users';

export {
  Home,
  Api,
  Search,
  Users,
  NotFound,
};
