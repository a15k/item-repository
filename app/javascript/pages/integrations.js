import {
  ListGroup, ListGroupItem,
} from 'reactstrap';
import Button from '../components/button';
import { React, observer, observable, action, computed } from '../helpers/react';
import Assessment from '../models/assessment';
import Interactions from '../models/interaction-app';
import SwipeableViews from 'react-swipeable-views';
import Listing from './integrations/listing';
import Edit from './integrations/edit';

@observer
export default class Integrations extends React.Component {

  componentDidMount() {
    Interactions.collection.api.fetch()
      .then(() => {
        this.editingIntegration = Interactions.collection.array[0]

      });
  }

  @observable editingIntegration;

  @computed get currentIndex() {
    return this.editingIntegration ? 1 : 0;
  }

  @action.bound onAdd() {
    Interactions.collection.create({});
  }

  @action.bound onEdit(interaction) {
    this.editingIntegration = interaction;
  }

  @action.bound onEditComplete() {
    this.editingIntegration = null;
  }

  @action.bound onDelete(ev) {
    Interactions.collection.destroy({
      id: ev.currentTarget.parentElement.dataset.id,
    });
  }

  render() {
    return (
      <div className="integrations">
        <h1>integrations</h1>
        <SwipeableViews index={this.currentIndex}>
          <Listing onEdit={this.onEdit} />
          <Edit app={this.editingIntegration} onDone={this.onEditComplete} />
        </SwipeableViews>
        <Button icon="plus" onClick={this.onAdd}>Add</Button>
      </div>
    );
  }

}
