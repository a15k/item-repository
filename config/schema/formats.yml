---
"$schema": http://json-schema.org/draft-04/hyper-schema
title: Formats accepted by the A15K item repository
description: A format describes how an accessmentâ€˜s question and solution are encoded.
stability: prototype
strictProperties: true
type:
- object
definitions:
  name:
    description: The unique name of format
    readOnly: true
    type:
    - string
  description:
    description: A human readable description of a format
    readOnly: true
    type:
    - string
  identity:
    anyOf:
      -
        "$ref": "/formats#/definitions/id"

links:
- description: List all formats.
  http_header:
    Authorization: <jwt-access-token>
  href: "/formats"
  method: GET
  rel: instances
  rel: instances
  title: List

properties:
  name:
    "$ref": "/formats#/definitions/name"
  description:
    "$ref": "/formats#/definitions/description"
id: schema/formats
