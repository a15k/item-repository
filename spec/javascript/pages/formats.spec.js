import renderer from 'react-test-renderer';
import { MemoryRouter as R } from 'react-router';
import { Formats } from 'pages/formats';
import FormatModel from 'models/format';
import Collection from 'models/model-collection';
import Factory from '../factories';

jest.mock('models/iframe-settings', () => ({
  assets: { javascript: '/preview.js', stylesheet: 'preview.css' },
}));

describe('Formats Page', () => {
  let props;

  beforeEach(() => {
    const collection = new Collection(FormatModel);
    collection.fromJSON([ Factory.create('Format') ]);
    collection.fromJSON([{
      id: '1234', name: 'Test Format',
      specification: 'This is <b>the best</b> format',
    }]);
    collection.api.requestCounts.get +=1;
    props = {
      match: { params: {} },
      collection,
      history: { push: jest.fn() },
    };
  });

  it('renders formats listing', () => {
    const search = renderer.create(<R><Formats {...props} /></R>);
    expect(search.toJSON()).toMatchSnapshot();
    search.unmount();
  });

  it('renders a format spec', () => {
    props.match.params.id = props.collection.array[0].id;
    const search = renderer.create(<Formats {...props} />);
    expect(search.toJSON()).toMatchSnapshot();
    search.unmount();
  });

});
