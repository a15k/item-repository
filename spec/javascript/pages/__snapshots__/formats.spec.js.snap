// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formats Page renders a format spec 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .icon + span {
  margin-left: 0.5rem;
}

<div>
  <h3>
    <button
      className="c0 btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 icon"
        data-icon="chevron-left"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Formats listing
      </span>
    </button>
    Kiehn - Ankunding
  </h3>
  <iframe
    id="format-specification-379d79cc-cd42-443c-9f2e-15087a5bc518"
    src="/assessment/format/379d79cc-cd42-443c-9f2e-15087a5bc518"
    srcDoc="
  <!DOCTYPE html>
  <html>
    <head>
      <title>Format Specification for illo</title>
      <link rel=\\"stylesheet\\" media=\\"all\\" href=\\"undefined\\">
      <script src=\\"/preview.js\\"></script>
    </head>
    <body>
      undefined
    </body>
  </html>
"
    style={
      Object {
        "border": "0px",
        "minWidth": "100%",
        "width": "1px",
      }
    }
  />
</div>
`;

exports[`Formats Page renders formats listing 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .icon + span {
  margin-left: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > *:last-child {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
}

<div
  className="formats-page"
>
  <h3>
    Formats
  </h3>
  <ul
    className="list-group"
  >
    <li
      className="c0 list-group-item"
    >
      <span>
        Kiehn - Ankunding
      </span>
      <span>
        379d79cc-cd42-443c-9f2e-15087a5bc518
      </span>
      <button
        className="c1 btn btn-secondary"
        data-id="379d79cc-cd42-443c-9f2e-15087a5bc518"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-right fa-w-10 icon"
          data-icon="chevron-right"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
    <li
      className="c0 list-group-item"
    >
      <span>
        Test Format
      </span>
      <span>
        1234
      </span>
      <button
        className="c1 btn btn-secondary"
        data-id="1234"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-right fa-w-10 icon"
          data-icon="chevron-right"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </li>
  </ul>
</div>
`;
